### PR TITLE
[Security] RememberMeFactory: refactoring (extracted methods)

### PR DESCRIPTION
**Pull request by MacDada**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | ~ |
| License | MIT |
| Doc PR | ~ |

It's easier to read and maintain code, when methods are short and with names that make the code self-describing.

### CODE REVIEW

File: src/Symfony/Bundle/SecurityBundle/DependencyInjection/Security/Factory/RememberMeFactory.php [hunk 1]
```diff
@@ -68,8 +131,29 @@ public function create(ContainerBuilder $container, $id, $config, $userProvider,
...
 
-        // attach to remember-me aware listeners
+        $rememberMeServices->replaceArgument(0, $this->findUserProviders($container, $id, $config));
+    }
+
+    private function findUserProviders(ContainerBuilder $container, $id, array $config)
+    {
+        $userProviders = $this->findAndAttachToRememberMeAwareListeners($container, $id, $config);
+
+        if ($config['user_providers']) {
```
Comment (MacDada) [inline src/Symfony/Bundle/SecurityBundle/DependencyInjection/Security/Factory/RememberMeFactory.php]: BTW, is it OK that `user_providers` from `$config` override remember me aware listeners? Or is that a bug?

File: src/Symfony/Bundle/SecurityBundle/DependencyInjection/Security/Factory/RememberMeFactory.php [hunk 2]
```diff
@@ -31,7 +31,68 @@ class RememberMeFactory implements SecurityFactoryInterface
...
+        foreach ($this->options as $name => $value) {
+            if (is_bool($value)) {
+                $builder->booleanNode($name)->defaultValue($value);
+            } else {
+                $builder->scalarNode($name)->defaultValue($value);
+            }
+        }
+    }
+
+    private function getRememberMeServicesId($id, array $config)
```
Comment (MacDada) [inline src/Symfony/Bundle/SecurityBundle/DependencyInjection/Security/Factory/RememberMeFactory.php]: I'm not sure if I should assume that `$config` is always array (here and in other methods). If not, I'll remove the typehints.

**Comment by weaverryan** [comment]
Hey @MacDada!

This is an easy win in theory, but it has *big* risks (did we accidentally change something) with only a small weird (internal refactoring). That's especially true because this class doesn't have a test with it. If you added a pretty thorough test that passed *before* the refactoring, that would definitely help. Someone, we need to make sure that the new code is exactly the same as the old, and obviously the diff is too big to be sure of that just by looking :).

Oh, and smaller changes are always better - it's easier for us to merge a number of subsequent smaller changes, because we can see that no BC breaks were introduced. I've had internal code refactorings like this not get accepted before because I just put too much into them and made the core team's review job impossible.

Cheers!

**Comment by MacDada** [comment]
@weaverryan As for the tests, well, I didn't even check if it's covered. I just run them and they passed :D I'm not feeling like writing them from scratch, but I can split this into smaller commits, so that review can be easier.

**Comment by weaverryan** [comment]
@MacDada I was actually thinking smaller pull requests, but this might in fact work - the first two commits were very easy to review. I'll review myself if I can find some time.

To others - if you review the individual commits, then we might be able to give this a +1.

Cheers!

**Comment by jakzal** [comment]
@MacDada @weaverryan I don't think we can merge this with no tests. I'll help and write them.

**Comment by MacDada** [comment]
> the first two commits were very easy to review

@weaverryan  Cut and paste into a function. Other type of changes are in separate commits, so they can be easily spotted :)

>  I'll help and write them [tests].

@jakzal Thx! They'll come handy at https://github.com/symfony/symfony/issues/14780 as well.

**Comment by fabpot** [comment]
Any news on this one?

**Comment by fabpot** [comment]
Closing as there is no feedback and that does not fix any bugs anyway.

**Pull request closed without merging**